src/pages/Login.js

```diff
@@ -51,6 +51,7 @@ const Login = () => {
 
   return (
     <div>
+{/* AKOS REVIEW NOTE: We won't ever see this welcome message, since we re-navigate to the / route in the header when the login is successful */}
       {userName && <p>Welcome {userName}!</p>}
       <h1>Login Form</h1>
```

Comment:
```suggestion
      <h1>Login Form</h1>
```

Reply:
Sorry I didn't catch you , would  you like to propose the deletion of this line  {userName && <p>Welcome {userName}!</p>}  and to leave just <h1>Login Form</h1> ?

Reply:
Yes exactly. 

Right now, after your login succeeds and the form is submitted, you're redirecting to the `/` route for the header. When you do that, the user won't even get to see this Welcome message, since the `Login` component will be navigated away from, and instead it's going to be showing the `Header` component (except everything under \<Router> on lines [34 to 42 in HeaderContent.js](https://github.com/akosasante/Shop-online/blob/auth/src/data/HeaderContent.js#L34-L42) will be empty, since none of those routes match `/`). 

If you want to show the Welcome message, you'll need to "lift the state up". Another way to think about it is, "Where do I want to show the welcome message?", "What components need to know about the logged-in user" ? 

The answers would be "I want to show the welcome message in the header section, even when the login component goes away (ie: even when we're on the `"/"` route instead of just in the `"/login"` route)". And "The header needs to know, and maybe the rest of the page needs to know eventually too" (ie: maybe when you get to making the cart and creating an order).

For now, let's just move the state up to the Header section:

```
// Login.js
import React, { useState } from 'react'
import { Link, Navigate } from 'react-router-dom'
import { handleLogin } from '../utils/api'
const rootUrl = 'https://node-course-e-commerce-8r2s.onrender.com'

// The parent component will pass in a function via the props, called afterLogin, that this child
// component can call when it's done logging in; to tell the parent that login succeeded and pass up the user object
const Login = ({ afterLogin }) => {
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const [redirectToHome, setRedirectToHome] = useState(false)

  const handleSubmit = async (e) => {
    e.preventDefault()
    if (!email || !password) return
    const user = { email, password }
    const success = await handleLogin(user) // Update handleLogin in api.js to return the user in an object;  instead of just returning true
    if (success) {
      setEmail('')
      setPassword('')
      afterLogin(success.user)
      setRedirectToHome(true)
    }
  }

  if (redirectToHome) {
    return <Navigate to='/' />
  }

  return (
    <div>
      <h1>Login Form</h1>
      <form onSubmit={handleSubmit}>
        <div>
          <label>Email</label>
          <input
            type='email'
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
        </div>
        <div>
          <label>Password</label>
          <input
            type='password'
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
        </div>
        <button type='submit'>Login</button>
      </form>
      <p>
        Don't have an account? <Link to='/register'>Register</Link>
      </p>
    </div>
  )
}

export default Login

// HeaderContent.js
import React, { useEffect, useRef } from 'react'
import { gsap } from 'gsap'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'

import Login from '../pages/Login'
import Register from '../pages/Register'
import Logout from '../pages/Logout'

const Header = () => {
  let text = useRef(null)
  const [userName, setUserName] = useState(undefined)

  useEffect(() => {
    gsap.to(text, {
      duration: 8,
      delay: 3,
      opacity: 1,
      repeat: -1
    })
  }, [])
  
  function setUserNameAfterLogin(user) {
    setUserName(user.name)
  }

  return (
    <div className='header_container'>
      <img className='logoImg' src='Logo.png' alt='logo' />

      <p
        ref={(el) => {
          text = el
        }}
        className='par'
      >
        Benvenuto su Jewelry Shop Olivia Rossi!
      </p>
      <Router>
        <Routes>
          <Route exact path='/login' element={<Login afterLogin={setUserNameAfterLogin} />} />

          <Route exact path='/register' element={<Register />} />

          <Route exact path='/logout' element={<Logout />} />
          
          <Route path='*'>
            {/* This wildcard route will be rendered for anything that didn't match the above three routes /*}
            {{userName && <p>Welcome {userName}!</p>}
          </Route>
        </Routes>
      </Router>

      <a href='##' className='carello-button'>
        <img className='carelloImg' src='Carello.png' alt='logo' />
      </a>
    </div>
  )
}

export default Header
```

Reply:
https://react.dev/learn/sharing-state-between-components